src/main/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiService.scala

```diff
@@ -22,6 +26,27 @@ trait EntityApiService extends HttpService with PerRequestCreator with UserInfoD
   val workspaceServiceConstructor: UserInfo => WorkspaceService
 
   val entityRoutes = requireUserInfo() { userInfo =>
+    path("workspaces" / Segment / Segment / "entityQuery" / Segment) { (workspaceNamespace, workspaceName, entityType) =>
+      get {
+        parameters('page.?, 'pageSize.?, 'sortField.?, 'sortDirection.?, 'filterTerms.?) { (page, pageSize, sortField, sortDirection, filterTerms) =>
+          val toIntTries = Map("page" -> page, "pageSize" -> pageSize).map { case (k,s) => k -> Try(s.map(_.toInt)) }
+          val sortDirectionTry = sortDirection.map(dir => Try(SortDirections.fromString(dir))).getOrElse(Success(Ascending))
+
+          val errors = toIntTries.collect {
+            case (k, Failure(t)) => s"$k must be a positive integer"
+            case (k, Success(Some(i))) if i <= 0 => s"$k must be a positive integer"
+          } ++ (if (sortDirectionTry.isFailure) Seq(sortDirectionTry.failed.get.getMessage) else Seq.empty)
+
+          if (errors.isEmpty) {
+            val entityQuery = EntityQuery(page.map(_.toInt).getOrElse(1), pageSize.map(_.toInt).getOrElse(10), sortField.getOrElse("name"), sortDirectionTry.get, filterTerms)
```

Comment:
Still want an answer on whether `toIntTries("page").getOrElse(1)` etc is clearer / more efficient here